srtcore/sync_posix.cpp

```diff
@@ -114,8 +114,18 @@ int64_t get_cpu_frequency()
     return frequency;
 }
 
-const int64_t s_cpu_frequency = get_cpu_frequency();
+int caclulate_subsecond_signs(int64_t ticks_per_us)
+{
+    int signs = 0;
+    while (ticks_per_us /= 10) ++signs;
```

Comment:
I think you could speed this up by using "binary search" method and start from the least value that you expect to have in all implementations (I guess it's 6). And the better name for it is "precission".

Reply:
This function is called only once, and the value is then stored in `s_clock_subsecond_signs`.
No need for optimization.

Reply:
Ah, there's no problem then.

Still, I think `calculate_subsecond_prec` should be better and `clock_subsecond_precision()` is a good enough name, I'd only be picky about the naming convention (`clockSubsecondPrecision()`).

Reply:
I have already renamed `aclulate_subsecond_signs(..)`  to `count_subsecond_precision(..)`. Not sure why GitHub still shows the old naming here.

Reply:
Renamed `clock_subsecond_precision()` to `clockSubsecondPrecision()`